ipa-core/src/utils/arraychunks.rs

```diff
@@ -17,17 +17,17 @@ pub struct ArrayChunk<I, const L: usize> {
     iter: I,
 }
 
-impl<I: Iterator<Item = Fp61BitPrime>> ArrayChunkIterator for I {}
+impl<F: PrimeField, I: Iterator<Item = F>> ArrayChunkIterator for I {}
 
 #[allow(clippy::while_let_on_iterator)]
-impl<I, const L: usize> Iterator for ArrayChunk<I, L>
+impl<F: PrimeField, I, const L: usize> Iterator for ArrayChunk<I, L>
 where
-    I: Iterator<Item = Fp61BitPrime>,
+    I: Iterator<Item = F>,
```

Comment:
for consistency, it is better to specify all trait bounds either in `where` clause or next to `impl`. 
In this case, I think you should just move everything to `where`